sdks/common/runner-api/src/main/proto/beam_job_api.proto

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Protocol Buffers describing the Job API, api for communicating with a runner
+ * written in language different from the SDK language over GRPC.
+ */
+
+syntax = "proto3";
+
+package org.apache.beam.runner_api.v1;
+
+option java_package = "org.apache.beam.common.job";
+option java_outer_classname = "JobServiceApi";
+
+import "beam_runner_api.proto";
+import "google/protobuf/struct.proto";
+
+
+// Job Service for running RunnerAPI pipelines
+service JobService {
+  rpc run (SubmitJobRequest) returns (SubmitJobResponse) {}
+  rpc getState (GetJobStateRequest) returns (GetJobStateResponse) {}
+  rpc cancel (CancelJobRequest) returns (CancelJobResponse) {}
+  rpc getStateStream (GetJobStateRequest) returns (stream GetJobStateResponse) {}
+  rpc getMessageStream (JobMessagesRequest) returns (stream JobMessagesResponse) {}
+}
+
+
+// Submit is a synchronus request that returns a jobId back
+// GRPC will throw a GRPC_STATUS_UNAVAILABLE if server is down
+// We'll throw an error with code ALREADY_EXISTS if the jobName is reused
+// We'll throw an error UNKNOWN for all other messages
+message SubmitJobRequest {
+  org.apache.beam.runner_api.v1.Pipeline pipeline = 1; // (required)
+  google.protobuf.Struct pipelineOptions = 2; // (required)
+  string jobName = 3;  // (required)
+}
+
+message SubmitJobResponse {
+  string jobId = 1; // (required)
+}
+
+
+
+// Cancel is a synchronus request that returns a jobState back
+// GRPC will throw a GRPC_STATUS_UNAVAILABLE if server is down
+// We'll throw an error with code NOT_FOUND if the jobId is not found
+message CancelJobRequest {
+  string jobId = 1; // (required)
+
+}
+
+message CancelJobResponse {
+  JobState.JobStateType state = 1; // (required)
+}
+
+
+// GetState is a synchronus request that returns a jobState back
```

Comment:
There are two APIs that use the same request format one sends a single response back and the other allows you to subscribe to state changes.